src/Features/Core/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.cs

```diff
@@ -22,111 +26,200 @@ public DiagnosticIncrementalAnalyzer(DiagnosticAnalyzerService owner, int correl
         }
 
         #region IIncrementalAnalyzer
-        public override Task AnalyzeDocumentAsync(Document document, SyntaxNode bodyOpt, CancellationToken cancellationToken)
+        public async override Task AnalyzeDocumentAsync(Document document, SyntaxNode body, CancellationToken cancellationToken)
         {
+            // ToDo: Should there be an exception handler here? Should there be checks for cancellation?
```

Comment:
having exception handler at the same level as "await" gives better crash report. if you don't care, outter one already have exception handler.